src/abstractdataframe/show.jl

```diff
@@ -55,22 +55,48 @@ end
 
 """Return compact string representation of type T"""
 function compacttype(T::Type, maxwidth::Int=8)
+    maxwidth = max(8, maxwidth)
+
     T === Any && return "Any"
     T === Missing && return "Missing"
+
     sT = string(T)
-    length(sT) ≤ maxwidth && return sT
+    textwidth(sT) ≤ maxwidth && return sT
+
     if T >: Missing
         T = nonmissingtype(T)
         sT = string(T)
-        suffix = "⍰"
-        # handle the case when after removing Missing union type name is short
-        length(sT) ≤ 8 && return sT * suffix
+        suffix = "?"
+        maxwidth -= 1 # we will add "?" at the end
```

Comment:
Previously we didn't include `⍰` in the count, right? I think we'd better not include it, as it means that narrow columns allowing and not allowing for missing values will print a different abbreviation for the type, e.g. `DataFrame(x=BigFloat[1, 2], y=Union{BigFloat, Missing}[1, 2])` will use "BigFloa" and "BigFlo" (IIUC).

Also, by taking into account `?` and `…` the number of characters used for the type is now 6 instead of 8 when it's too long and allows for missing values. Previously `BigFloat` would always be printed in full, and now it isn't. Is this intentional?

Reply:
Well - it was intentional (to normalize everything to 8), but I can change it. I will switch not to count "?" when determining width (which means sometimes it will be 9 like for `BigFloat?`).